### PR TITLE
refactor: share components/flows API

### PR DESCRIPTION
**Pull request by igorrCarvalho**
This PR focuses on refactoring the sharing components and flows API to enhance code maintainability, readability, and scalability.

### CODE REVIEW

File: src/frontend/src/modals/shareModal/index.tsx [hunk 1]
```diff
@@ -63,6 +67,47 @@ export default function ShareModal({
   const saveFlow = useFlowsManagerStore((state) => state.saveFlow);
   const { data, isFetching } = useGetTagsQuery();
 
+  function successShare() {
```
Comment (anovazzi1) [inline src/frontend/src/modals/shareModal/index.tsx:70]: is there any reason to save the flow again after sharing? I saw in the file history that it's already like that, but I think this code was wrong before and we could remove it in this refactor

**Review by anovazzi1 — CHANGES_REQUESTED** [review]
LGTM

File: src/frontend/src/modals/shareModal/index.tsx (hunk 1 above)
Comment (igorrCarvalho) [reply src/frontend/src/modals/shareModal/index.tsx:69]: I think it saves the flow immediately after sharing, so it already appears in the sidebar. This way, the user doesn't have to go to the store to find the flow in the 'Saved' section. but ill check

**Review by anovazzi1 — APPROVED** [review]
LGTM

**Pull request closed without merging**